projects/exui-common-lib/src/lib/components/generic-filter/generic-filter.component.ts

```diff
@@ -55,96 +59,182 @@ export class GenericFilterComponent {
     this._settings = value;
   }
 
-  public applyFilter(): void {
-    if (!this.checkFieldsConstraints(this.selected)) {
-      this.filterService.givenErrors.next(null);
-      this.filterService.persist(this.settings, this.config.persistence);
+  public ngOnInit(): void {
+    if (!this.settings) {
+      this.getSettings();
     }
+    this.mergeDefaultFields(this.settings);
+    this.buildForm(this.config, this.settings);
+    this.formSub = this.form.valueChanges.subscribe(() => this.submitted = false);
   }
 
-  public cancelFilter(): void {
-    const cancelFields = JSON.parse(JSON.stringify(this.config.cancelSetting.fields));
-    this._settings.fields = cancelFields;
-    this.selected = cancelFields;
-    this.applyFilter();
-  }
-
-  public isSelected(id: string, key: string): boolean {
-    if (this.settings) {
-      return this.settings.fields.some(x => x.name === id) && this.settings.fields.some(x => x.value.some(v => v === key));
-    }
-    return false;
-  }
-
-  public onSelectionChange(event: Event): void {
-    const input = event.target as HTMLInputElement;
-    const fieldName = input.getAttribute('field');
-    const match = this.selected.find(x => x.name === fieldName);
-    if (match) {
-      if (input.type === 'checkbox') {
-        if (input.checked) {
-          match.value.push(input.value);
-        } else {
-          const index: number = match.value.indexOf(input.value);
-          if (index !== -1) {
-            match.value.splice(index, 1);
-          }
-        }
+  public ngOnDestroy(): void {
+    if (this.formSub) {
+      this.formSub.unsubscribe();
+    }
+  }
+
+  public hidden(field: FilterFieldConfig, form: FormGroup): boolean {
+    if (!field.showCondition) {
+      return false;
+    }
+    if (typeof field.showCondition === 'string') {
+      const control = form.get(field.name) as FormControl;
+      const [name, value] = field.showCondition.split('=');
+      if (form.value && form.value[name] === value) {
+        control.setValidators(Validators.required);
```

Comment:
Are fields automatically required when they have a show condition? Not sure about that

Reply:
In all the cases we have as for now. Fields with a `showCondition` will be required once activated.